fdbrpc/AsyncFileCached.actor.h

```diff
@@ -52,19 +52,26 @@ struct EvictablePage {
 struct EvictablePageCache : ReferenceCounted<EvictablePageCache> {
 	using List = bi::list< EvictablePage, bi::member_hook< EvictablePage, bi::list_member_hook<>, &EvictablePage::member_hook>>;
 
-	EvictablePageCache() : pageSize(0), maxPages(0)	{
+	EvictablePageCache() : pageSize(0), maxPages(0) {}
+
+	explicit EvictablePageCache(int pageSize, int64_t maxSize) : pageSize(pageSize), maxPages(maxSize / pageSize) {
+		std::string cep = FLOW_KNOBS->CACHE_EVICTION_POLICY;
+		std::transform(cep.begin(), cep.end(), cep.begin(), ::tolower);
+		if (0 == cep.compare("random"))
```

Comment:
An exception should be thrown if the eviction type is not recognized, but also it would be good to error out as soon as possible from fdbserver if the knob is set incorrectly.  

To that end, I suggest making a public static helper function to convert a policy string to an enum, throwing an exception on failure (probably a new exception type).  That function can then be used here, and we can also call it from fdbserver.cpp using the knob's current value after all of the knob arguments would have been processed and set.

Minor style comment - Personally I think `cep == "random"` is a bit more natural to read than `0 == cep.compare("random")`.

Reply:
@satherton thanks for the input. I have updated the patch with your suggestions.